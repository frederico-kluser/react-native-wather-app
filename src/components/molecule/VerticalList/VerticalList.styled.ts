import styled from 'styled-components/native';

const Container = styled.View`
  top: 62px;
`;

export default Container;
