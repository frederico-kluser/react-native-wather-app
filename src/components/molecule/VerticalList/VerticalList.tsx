import React from 'react';
import LargeItem from '../../atom/LargeItem/LargeItem';
import Container from './VerticalList.styled';

const VerticalList = ({nextDays}: any) => (
  <Container>
    {nextDays.map(
      ({dt, temp, weather}: any, index) =>
        index < 7 && (
          <LargeItem
            key={dt}
            text={dt}
            min={temp.min}
            max={temp.max}
            icon={weather[0].description}
          />
        ),
    )}
  </Container>
);

export default VerticalList;
