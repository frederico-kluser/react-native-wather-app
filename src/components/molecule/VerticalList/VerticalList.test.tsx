import React from 'react';
import renderer from 'react-test-renderer';
import VerticalList from './VerticalList';

const mockCity = {
  city: 'New York',
  country: 'US',
  description: 'New York, NY, USA',
  favorite: false,
  humidity: 54,
  maxTemperature: 8,
  minTemperature: 3,
  nextDays: [
    {
      dt: 1644339600,
      sunrise: 1644321522,
      sunset: 1644358908,
      moonrise: 1644335580,
      moonset: 1644298260,
      moon_phase: 0.25,
      temp: {
        day: 279.83,
        min: 273.37,
        max: 280.11,
        night: 273.37,
        eve: 275.69,
        morn: 274.95,
      },
      feels_like: {
        day: 276.71,
        night: 268.94,
        eve: 270.82,
        morn: 270.73,
      },
      pressure: 1012,
      humidity: 54,
      dew_point: 271.42,
      wind_speed: 6.56,
      wind_deg: 259,
      wind_gust: 10.65,
      weather: [
        {
          id: 500,
          main: 'Rain',
          description: 'light rain',
          icon: '10d',
        },
      ],
      clouds: 1,
      pop: 0.89,
      rain: 0.74,
      uvi: 2.46,
    },
    {
      dt: 1644426000,
      sunrise: 1644407853,
      sunset: 1644445382,
      moonrise: 1644423780,
      moonset: 1644388380,
      moon_phase: 0.28,
      temp: {
        day: 277.51,
        min: 273,
        max: 279.97,
        night: 275.95,
        eve: 277.75,
        morn: 273.08,
      },
      feels_like: {
        day: 274.78,
        night: 272.77,
        eve: 274.56,
        morn: 269.67,
      },
      pressure: 1018,
      humidity: 48,
      dew_point: 267.47,
      wind_speed: 4.21,
      wind_deg: 174,
      wind_gust: 10.01,
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'clear sky',
          icon: '01d',
        },
      ],
      clouds: 0,
      pop: 0,
      uvi: 2.6,
    },
    {
      dt: 1644512400,
      sunrise: 1644494183,
      sunset: 1644531857,
      moonrise: 1644512280,
      moonset: 1644478440,
      moon_phase: 0.31,
      temp: {
        day: 280.78,
        min: 274.98,
        max: 281.98,
        night: 275.91,
        eve: 279.25,
        morn: 275.18,
      },
      feels_like: {
        day: 277.1,
        night: 271.28,
        eve: 275.15,
        morn: 272.26,
      },
      pressure: 1010,
      humidity: 54,
      dew_point: 271.95,
      wind_speed: 7.19,
      wind_deg: 264,
      wind_gust: 12.51,
      weather: [
        {
          id: 802,
          main: 'Clouds',
          description: 'scattered clouds',
          icon: '03d',
        },
      ],
      clouds: 45,
      pop: 0,
      uvi: 2.11,
    },
    {
      dt: 1644598800,
      sunrise: 1644580511,
      sunset: 1644618331,
      moonrise: 1644601080,
      moonset: 1644568440,
      moon_phase: 0.34,
      temp: {
        day: 279.04,
        min: 273.37,
        max: 280.68,
        night: 277.93,
        eve: 278.58,
        morn: 273.37,
      },
      feels_like: {
        day: 275.79,
        night: 275,
        eve: 275.45,
        morn: 269.53,
      },
      pressure: 1020,
      humidity: 39,
      dew_point: 266.04,
      wind_speed: 5.55,
      wind_deg: 264,
      wind_gust: 11.33,
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'clear sky',
          icon: '01d',
        },
      ],
      clouds: 6,
      pop: 0,
      uvi: 2.12,
    },
    {
      dt: 1644685200,
      sunrise: 1644666837,
      sunset: 1644704804,
      moonrise: 1644690420,
      moonset: 1644658080,
      moon_phase: 0.37,
      temp: {
        day: 286.33,
        min: 277.56,
        max: 286.68,
        night: 279.28,
        eve: 282.08,
        morn: 278.69,
      },
      feels_like: {
        day: 285.09,
        night: 275.78,
        eve: 279.43,
        morn: 275.43,
      },
      pressure: 1014,
      humidity: 53,
      dew_point: 276.7,
      wind_speed: 5.89,
      wind_deg: 233,
      wind_gust: 13.61,
      weather: [
        {
          id: 802,
          main: 'Clouds',
          description: 'scattered clouds',
          icon: '03d',
        },
      ],
      clouds: 25,
      pop: 0,
      uvi: 3,
    },
    {
      dt: 1644771600,
      sunrise: 1644753163,
      sunset: 1644791278,
      moonrise: 1644780060,
      moonset: 1644747540,
      moon_phase: 0.4,
      temp: {
        day: 271.54,
        min: 269.66,
        max: 273.11,
        night: 269.66,
        eve: 269.92,
        morn: 270.55,
      },
      feels_like: {
        day: 266.07,
        night: 263.2,
        eve: 263.2,
        morn: 263.96,
      },
      pressure: 1021,
      humidity: 38,
      dew_point: 259.04,
      wind_speed: 7.13,
      wind_deg: 322,
      wind_gust: 9.99,
      weather: [
        {
          id: 600,
          main: 'Snow',
          description: 'light snow',
          icon: '13d',
        },
      ],
      clouds: 100,
      pop: 0.37,
      snow: 0.18,
      uvi: 3,
    },
    {
      dt: 1644858000,
      sunrise: 1644839487,
      sunset: 1644877751,
      moonrise: 1644870120,
      moonset: 1644836520,
      moon_phase: 0.44,
      temp: {
        day: 270.94,
        min: 267.8,
        max: 272.3,
        night: 272.3,
        eve: 272.17,
        morn: 267.8,
      },
      feels_like: {
        day: 268.72,
        night: 268.19,
        eve: 268.93,
        morn: 261.61,
      },
      pressure: 1021,
      humidity: 27,
      dew_point: 254.11,
      wind_speed: 5.35,
      wind_deg: 17,
      wind_gust: 6.71,
      weather: [
        {
          id: 803,
          main: 'Clouds',
          description: 'broken clouds',
          icon: '04d',
        },
      ],
      clouds: 65,
      pop: 0,
      uvi: 3,
    },
    {
      dt: 1644944400,
      sunrise: 1644925809,
      sunset: 1644964224,
      moonrise: 1644960360,
      moonset: 1644925140,
      moon_phase: 0.47,
      temp: {
        day: 270.07,
        min: 265.91,
        max: 271.96,
        night: 265.91,
        eve: 266.92,
        morn: 271.13,
      },
      feels_like: {
        day: 263.07,
        night: 258.91,
        eve: 259.92,
        morn: 265.22,
      },
      pressure: 1027,
      humidity: 27,
      dew_point: 253.3,
      wind_speed: 9.39,
      wind_deg: 312,
      wind_gust: 14.2,
      weather: [
        {
          id: 803,
          main: 'Clouds',
          description: 'broken clouds',
          icon: '04d',
        },
      ],
      clouds: 66,
      pop: 0,
      uvi: 3,
    },
  ],
  status: 'clear sky',
  temperature: 6,
  wind: 6,
  localFavorite: true,
};

test('renders correctly', () => {
  const tree = renderer
    .create(<VerticalList nextDays={mockCity.nextDays} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
