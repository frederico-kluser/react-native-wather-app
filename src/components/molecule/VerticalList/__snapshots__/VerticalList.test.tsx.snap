// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "top": "8%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": "6%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "fontWeight": "500",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      quarta
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/rain.png",
        }
      }
      style={
        Array [
          Object {
            "height": 32,
            "left": "47%",
            "position": "absolute",
            "top": -8,
            "width": 32,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 13,
            "fontWeight": "bold",
            "left": "80%",
            "position": "absolute",
          },
        ]
      }
    >
      0
      °
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "blue",
            "fontSize": 12,
            "fontWeight": "bold",
            "position": "absolute",
            "right": "0%",
          },
        ]
      }
    >
      6
      °
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": "6%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "fontWeight": "500",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      quinta
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/sun.png",
        }
      }
      style={
        Array [
          Object {
            "height": 32,
            "left": "47%",
            "position": "absolute",
            "top": -8,
            "width": 32,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 13,
            "fontWeight": "bold",
            "left": "80%",
            "position": "absolute",
          },
        ]
      }
    >
      0
      °
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "blue",
            "fontSize": 12,
            "fontWeight": "bold",
            "position": "absolute",
            "right": "0%",
          },
        ]
      }
    >
      6
      °
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": "6%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "fontWeight": "500",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      sexta
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/cloud.png",
        }
      }
      style={
        Array [
          Object {
            "height": 32,
            "left": "47%",
            "position": "absolute",
            "top": -8,
            "width": 32,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 13,
            "fontWeight": "bold",
            "left": "80%",
            "position": "absolute",
          },
        ]
      }
    >
      1
      °
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "blue",
            "fontSize": 12,
            "fontWeight": "bold",
            "position": "absolute",
            "right": "0%",
          },
        ]
      }
    >
      8
      °
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": "6%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "fontWeight": "500",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      sábado
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/sun.png",
        }
      }
      style={
        Array [
          Object {
            "height": 32,
            "left": "47%",
            "position": "absolute",
            "top": -8,
            "width": 32,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 13,
            "fontWeight": "bold",
            "left": "80%",
            "position": "absolute",
          },
        ]
      }
    >
      0
      °
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "blue",
            "fontSize": 12,
            "fontWeight": "bold",
            "position": "absolute",
            "right": "0%",
          },
        ]
      }
    >
      7
      °
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": "6%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "fontWeight": "500",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      domingo
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/cloud.png",
        }
      }
      style={
        Array [
          Object {
            "height": 32,
            "left": "47%",
            "position": "absolute",
            "top": -8,
            "width": 32,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 13,
            "fontWeight": "bold",
            "left": "80%",
            "position": "absolute",
          },
        ]
      }
    >
      4
      °
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "blue",
            "fontSize": 12,
            "fontWeight": "bold",
            "position": "absolute",
            "right": "0%",
          },
        ]
      }
    >
      13
      °
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": "6%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "fontWeight": "500",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      segunda
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/sunSnow.png",
        }
      }
      style={
        Array [
          Object {
            "height": 32,
            "left": "47%",
            "position": "absolute",
            "top": -8,
            "width": 32,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 13,
            "fontWeight": "bold",
            "left": "80%",
            "position": "absolute",
          },
        ]
      }
    >
      -3
      °
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "blue",
            "fontSize": 12,
            "fontWeight": "bold",
            "position": "absolute",
            "right": "0%",
          },
        ]
      }
    >
      0
      °
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": "6%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "fontWeight": "500",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      terça
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/cloudWind.png",
        }
      }
      style={
        Array [
          Object {
            "height": 32,
            "left": "47%",
            "position": "absolute",
            "top": -8,
            "width": 32,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 13,
            "fontWeight": "bold",
            "left": "80%",
            "position": "absolute",
          },
        ]
      }
    >
      -5
      °
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "blue",
            "fontSize": 12,
            "fontWeight": "bold",
            "position": "absolute",
            "right": "0%",
          },
        ]
      }
    >
      0
      °
    </Text>
  </View>
</View>
`;
